src/js/game/LevelMVC/LevelPlane.js

```diff
@@ -683,6 +691,29 @@ module.exports = class LevelPlane {
     return this.getOrthogonalPositions(position).some(orthogonalPosition => {
       const block = this.getBlockAt(orthogonalPosition);
       if (block) {
+        if (this.getBlockAt(position).blockType.startsWith("piston")) {
+          let piston = this.getBlockAt(position);
+          let direction = piston.blockType.substring(6, 7);
+          let ignoreThisSide = [0, 0];
+          switch (direction) {
+            case "D":
+              ignoreThisSide = [0, 1];
+              break;
+            case "U":
+              ignoreThisSide = [0, -1];
+              break;
+            case "L":
+              ignoreThisSide = [-1, 0];
+              break;
+            case "R":
+              ignoreThisSide = [1, 0];
+              break;
+          }
```

Comment:
This switch statement is very nearly repeating [this switch statement](https://github.com/code-dot-org/craft/blob/master/src/js/game/LevelMVC/LevelPlane.js#L555-L568), and is also relying on some subtle properties of the blockType string; both of these instances should be moved into LevelBlock

Reply:
Yeah, good idea. That was a copy paste for functionality so I could get out of the office on time. Will move that over right now